Services/UserService/Commands/CreateUserCommand.cs

```diff
@@ -39,27 +39,49 @@ public async Task<bool> Execute(CreateUserRequest request)
             validator.ValidateAndThrow(request);
 
             string passwordHash = ShaHash.GetPasswordHash(request.Password);
-
-            var user = new User
+            User user;
+            UserAvatar userAvatar = null;
+            if (request.Avatar.Length == 0)
             {
-                Id = Guid.NewGuid(),
-                Birthday = request.Birthday,
-                FirstName = request.FirstName,
-                LastName = request.LastName
-            };
-
+                user = new User
+                {
+                    Id = Guid.NewGuid(),
+                    Birthday = request.Birthday,
+                    FirstName = request.FirstName,
+                    LastName = request.LastName
+                };
+            }
+            else
+            {
+                user = new User
+                {
+                    Id = Guid.NewGuid(),
+                    UserAvatarId = Guid.NewGuid(),
+                    Birthday = request.Birthday,
+                    FirstName = request.FirstName,
+                    LastName = request.LastName
+                };
+                userAvatar = new UserAvatar
+                {
+                    Id = user.UserAvatarId,
+                    Avatar = request.Avatar,
+                    TypeAvatar = request.AvatarType
+                };
+            }

```

Comment:
Can we make it easier? May be move checks inside the user initialization?

Reply:
Maybe, but I don’t know how. Could you explain how?